src/vs/workbench/browser/parts/compositePart.ts

```diff
@@ -173,46 +170,31 @@ export abstract class CompositePart<T extends Composite> extends Part {
 		});
 	}
 
-	protected createComposite(id: string, isActive?: boolean): TPromise<Composite> {
+	protected createComposite(id: string, isActive?: boolean): Composite {
 
 		// Check if composite is already created
 		for (let i = 0; i < this.instantiatedComposites.length; i++) {
 			if (this.instantiatedComposites[i].getId() === id) {
-				return TPromise.as(this.instantiatedComposites[i]);
+				return this.instantiatedComposites[i];
 			}
 		}
 
 		// Instantiate composite from registry otherwise
-		let compositeDescriptor = this.registry.getComposite(id);
+		const compositeDescriptor = this.registry.getComposite(id);
 		if (compositeDescriptor) {
-			let loaderPromise = this.compositeLoaderPromises[id];
-			if (!loaderPromise) {
-				let progressService = this.instantiationService.createInstance(WorkbenchProgressService, this.progressBar, compositeDescriptor.id, isActive);
-				let compositeInstantiationService = this.instantiationService.createChild(new ServiceCollection([IProgressService, progressService]));
-
-				loaderPromise = compositeInstantiationService.createInstance(compositeDescriptor).then((composite: Composite) => {
-					this.mapProgressServiceToComposite[composite.getId()] = progressService;
+			const progressService = this.instantiationService.createInstance(WorkbenchProgressService, this.progressBar, compositeDescriptor.id, isActive);
+			const compositeInstantiationService = this.instantiationService.createChild(new ServiceCollection([IProgressService, progressService]));
 
-					// Remember as Instantiated
-					this.instantiatedComposites.push(composite);
+			const composite = (<any>compositeInstantiationService).createInstance(compositeDescriptor); // TODO@Ben why?
+			this.mapProgressServiceToComposite[composite.getId()] = progressService;
```

Comment:
The `InstantiationService#createInstance`-method is only declared for `SyncDescriptor0-8`. Maybe that is the problem. In general, you could try to avoid the use of `SyncDescriptor` altogether and try to use just the ctor function.